src/iface/ethernet.rs

```diff
@@ -336,15 +441,18 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
             }
         }
 
-        if !self.has_ip_addr(ipv4_repr.dst_addr) {
-            // Ignore IP packets not directed at us.
+        if !self.has_ip_addr(ipv4_repr.dst_addr) && !self.has_ip_mcast_addr(ipv4_repr.dst_addr) {
```

Comment:
Should probably be extracted into a function like `keep_ethernet_frame`; we also currently ignore all IPv4 broadcast packets, which is I think a bug.